warehouse/models/mart/ntd_validation/fct_ntd_rr20_service_checks.py

```diff
@@ -0,0 +1,223 @@
+                    elif (value_lastyr != 0) and abs((value_lastyr - value_thisyr)/value_lastyr) >= threshold:
+                        result = "fail"
+                        check_name = f"{variable}"
+                        mode = mode
+                        description = (f"The {variable} for {mode} has changed from last year by {round(abs((value_lastyr - value_thisyr)/value_lastyr)*100, 1)}%; please provide a narrative justification.")                        
+                    else:
+                        result = "pass"
+                        check_name = f"{variable}"
+                        mode = mode
+                        description = ""
+
+                    output_line = {"Organization": agency,
+                           "name_of_check" : check_name,
+                                   "mode": mode,
+                            "value_checked": f"{this_year} = {value_thisyr}, {last_year} = {value_lastyr}",
+                            "check_status": result,
+                            "Description": description}
+                    output.append(output_line)
+        else:
+            logger.info(f"There is no data for {agency}")
+    checks = pd.DataFrame(output).sort_values(by="Organization")
+    return checks
+
+
+def model(dbt, session):
+    # Set up the logger object
+    logger = write_to_log('rr20_ftc_servicechecks_log.log')
+
+    this_year=datetime.datetime.now().year
+    last_year = this_year-1
+    this_date=datetime.datetime.now().date().strftime('%Y-%m-%d') #for suffix on Excel files
+
+    #Load data from BigQuery - pass in the dbt model that we draw from.
+    allyears = dbt.ref("int_ntd_rr20_service_ratios")
+    allyears = allyears.toPandas()
+
+    # Run validation checks
+    cph_checks = check_rr20_ratios(allyears, 'cost_per_hr', .30, this_year, last_year, logger)
+    mpv_checks = check_rr20_ratios(allyears, 'miles_per_veh', .20, this_year, last_year, logger)
+    vrm_checks = check_single_number(allyears, 'Annual_VRM', this_year, last_year, logger, threshold=.30)
+    frpt_checks = check_rr20_ratios(allyears, 'fare_rev_per_trip', .25, this_year, last_year, logger)
+    rev_speed_checks = check_rr20_ratios(allyears, 'rev_speed', .15, this_year, last_year, logger)
+    tph_checks = check_rr20_ratios(allyears, 'trips_per_hr', .30, this_year, last_year, logger)
+    voms0_check = check_single_number(allyears, 'VOMX', this_year, last_year, logger)
+
+    # Combine checks into one table
+    rr20_checks = pd.concat([cph_checks, mpv_checks, vrm_checks, 
+                             frpt_checks, rev_speed_checks, 
+                             tph_checks, voms0_check], 
+                             ignore_index=True).sort_values(by="Organization")
+    
+    ## Part 1: save Excel file to GCS
+    GCS_FILE_PATH_VALIDATED = f"gs://calitp-ntd-report-validation/validation_reports_{this_year}" 
```

Comment:
As noted in Slack, the intention for a Python model in dbt is *not* to output files to other locations but is [simply to perform operations on data models](https://docs.getdbt.com/docs/build/python-models#defining-a-python-model) (totally akin to SQL models). Exports should be orchestrated using Airflow

Reply:
moving over to a different script, will try to deploy with Airflow as suggested.